File: dart/src/n02_testdatengenerator/data_generator.dart
import 'dart:convert';
import 'dart:io';
import 'dart:math';

part 'data_generator_csv.dart';
part 'data_generator_json.dart';
part 'data_generator_xml.dart';

abstract class DataGenerator {
  final List<double> _dataSet = List.empty(growable: true);

  void createData(final double min, final double max) {
    final Random random = Random();
    for (int i = 0; i < 100; i++) {
      this._dataSet.add(random.nextDouble() * (max - min) + min);
    }
  }

  Future<void> saveData(final File file);
}

File: java/src/n02_testdatengenerator/DataGenerator.java
package n02_testdatengenerator;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public abstract class DataGenerator {
    protected List<Double> dataSet = new ArrayList<>();

    public void createData(double min, double max) {
        Random random = new Random();
        for (int i = 0; i < 100; i++) {
            this.dataSet.add(random.nextDouble(min, max));
        }
    }

    abstract void saveData(File file);
}
